src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/CreateReindexRequestHandler.cs

```diff
@@ -57,6 +62,13 @@ public async Task<CreateReindexResponse> Handle(CreateReindexRequest request, Ca
                 throw new JobConflictException(string.Format(Resources.OnlyOneResourceJobAllowed, reindexJobId));
             }
 
+            // We need to pull in latest search parameter updates from the data store before creating a reindex job.
+            // There could be a potential delay of <see cref="ReindexJobConfiguration.JobPollingFrequency"/> before
+            // search parameter updates on one instance propagates to other instances. If we store the reindex
+            // job with the old hash value in _searchParameterDefinitionManager.SearchParameterHashMap, then we will
+            // not detect the resources that need to be reindexed.
+            await _searchParameterOperations.GetAndApplySearchParameterUpdates(cancellationToken);
```

Comment:
>GetAndApplySearchParameterUpdates [](http://example.com/codeflow?start=45&length=33)

@feordin 
I've give short glance and it looks like there is no locking inside that function.

Which is a bit worriesome. Considering there is 3 places now which would call this method, one of them is on a loop, could it be an issue?

Reply:
That's a good point!

Reply:
It is also true that SearchParameterStatusManager updates fields on the parameters, which would also change state.  Perhaps we can have a quick discussion on the best solution.